web/pdf_viewer.js

```diff
@@ -383,8 +383,8 @@ var PDFViewer = (function pdfViewer() {
             return;
           }
           var getPagesLeft = pagesCount;
-          for (var pageNum = 1; pageNum <= pagesCount; ++pageNum) {
-            pdfDocument.getPage(pageNum).then(function(pageNum, pdfPage) {
+          for (let pageNum = 1; pageNum <= pagesCount; ++pageNum) {
```

Comment:
Unless I'm missing something, this shouldn't be a problem. Note that Babel converts a loop with `let` into a loop over a *function*, thus creating the necessary closure; see e.g. [this simple example](https://babeljs.io/repl/#?babili=false&evaluate=true&lineWrap=false&presets=es2015&targets=&browsers=&builtIns=false&debug=false&experimental=true&loose=false&spec=false&code=var%20pageCount%20%3D%2010%3B%0Afor%20(let%20pageNum%20%3D%201%3B%20pageNum%20%3C%3D%20pagesCount%3B%20%2B%2BpageNum)%20%7B%0A%20Promise.resolve(()%20%3D%3E%20%7B%0A%20console.log(pageNum)%3B%0A%20%7D)%3B%0A%7D).

The complete code, taken from the preview above, thus becomes:
```javascript
onePageRenderedCapability.promise.then(function () {
  if (_pdfjs.PDFJS.disableAutoFetch) {
    pagesCapability.resolve();
    return;
  }
  var getPagesLeft = pagesCount;

  var _loop = function _loop(_pageNum) {
    pdfDocument.getPage(_pageNum).then(function (pdfPage) {
      var pageView = _this._pages[_pageNum - 1];
      if (!pageView.pdfPage) {
        pageView.setPdfPage(pdfPage);
      }
      _this.linkService.cachePageRef(_pageNum, pdfPage.ref);
      if (--getPagesLeft === 0) {
        pagesCapability.resolve();
      }
    });
  };

  for (var _pageNum = 1; _pageNum <= pagesCount; ++_pageNum) {
    _loop(_pageNum);
  }
});
```